File: api/src/main/kotlin/com/github/larscom/gitlabcidashboard/project/model/Project.kt
package com.github.larscom.gitlabcidashboard.project.model

import com.fasterxml.jackson.annotation.JsonAlias

data class Project(
    val id: Long,
    val name: String,
    @JsonAlias("default_branch") val defaultBranch: String,
    val topics: Set<String>
)

File: src/models/project.ts
import { Pipeline } from './pipeline'

export interface Namespace {
  id: number
  name: string
}

export type ProjectId = number

export interface Project {
  id: ProjectId
  name: string
  default_branch: string
  web_url: string
  topics: string[]
  description?: string
  namespace?: Namespace
  latest_pipeline?: Pipeline
}
